jsonschema-core.xml

```diff
@@ -1803,24 +1815,25 @@
 ]]>
                         </artwork>
                         <postamble>
-                            Here we see that the canonical URI for that "additionalProperties"
-                            subschema is still valid, while the non-canonical URI with the fragment
-                            beginning with "#/items/$ref" now resolves to nothing.
+                            Here we see that the URI for the "additionalProperties" schema object
+                            that is relative to its resource's canonical URI is still valid,
+                            while the URI relative to the "items" schema object's URI no longer
+                            resolves to anything.
                         </postamble>
                     </figure>
                     <t>
                         Note also that "https://example.com/foo#/items" is valid in both
                         arrangements, but resolves to a different value.  This URI ends up
-                        functioning similarly to a retrieval URI for a resource.  While valid,
-                        examining the resolved value and either using the "$id" (if the value
-                        is a subschema), or resolving the reference and using the "$id" of the
-                        reference target, is preferable.
+                        functioning similarly to a retrieval URI for a resource.  While this URI
+                        is valid, it is more robust to use the "$id" of the embedded or referenced
+                        resource unless it is specifically desired to identify the object containing
+                        the "$ref" in the second (non-embedded) arrangement.
                     </t>
                     <t>
-                        An implementation MAY choose not to support addressing schemas
-                        by non-canonical URIs. As such, it is RECOMMENDED that schema authors only
-                        use canonical URIs, as using non-canonical URIs may reduce
-                        schema interoperability.
+                        An implementation MAY choose not to support addressing schema resource
+                        contents by URIs using a base other than the resource's canonical URI,
+                        plus a JSON Pointer fragment relative to that base.  Therefore, schema
+                        authors SHOULD NOT rely on such URIs, as using them may reduce interoperability.
```

Comment:
Yes, very much outside the scope, please have this discussion elsewhere (you filed an issue, IIRC?)  It would be best not to bring up these comments in PRs because often they touch of big discussions despite repeated "this is outside the scope" attempts to close it down.  Although obviously there's no problem with that on this comment right now :-)